pkg/controller.v2/jobcontroller/pod.go

```diff
@@ -0,0 +1,199 @@
+package jobcontroller
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/controller"
+
+	jclogger "github.com/kubeflow/tf-operator/pkg/logger"
+)
+
+// When a pod is created, enqueue the job that manages it and update its expectations.
+func (jc *JobController) AddPod(obj interface{}) {
+	pod := obj.(*v1.Pod)
+	if pod.DeletionTimestamp != nil {
+		// on a restart of the controller controller, it's possible a new pod shows up in a state that
+		// is already pending deletion. Prevent the pod from being a creation observation.
+		// tc.deletePod(pod)
+		return
+	}
+
+	// If it has a ControllerRef, that's all that matters.
+	if controllerRef := metav1.GetControllerOf(pod); controllerRef != nil {
+		job := jc.resolveControllerRef(pod.Namespace, controllerRef)
+
+		logger := jclogger.LoggerForPod(pod, jc.Controller.GetAPIGroupVersionKind().Kind)
+
+		if job == nil {
+			logger.Info("This pod's job does not exists")
+			return
+		}
+
+		jobKey, err := controller.KeyFunc(job)
+		if err != nil {
+			logger.Infof("Failed to get the jobkey: %v", err)
+			return
+		}
+
+		if _, ok := pod.Labels[jc.Controller.GetReplicaTypeLabel()]; !ok {
+			logger.Infof("This pod maybe not created by %v", jc.Controller.GetControllerName())
+			return
+		}
+
+		rtype := pod.Labels[jc.Controller.GetReplicaTypeLabel()]
+		expectationPodsKey := GenExpectationPodsKey(jobKey, rtype)
+
+		jc.Expectations.CreationObserved(expectationPodsKey)
+		// TODO: we may need add backoff here
+		jc.WorkQueue.Add(jobKey)
+
+		return
+	}
+
+	// Otherwise, it's an orphan. Get a list of all matching controllers and sync
+	// them to see if anyone wants to adopt it.
+	// DO NOT observe creation because no controller should be waiting for an
+	// orphan.
+	// for _, tfjob := range tc.getPodJobs(pod) {
+	// 	tc.enqueueTFJob(tfjob)
+	// }
```

Comment:
Why do we need these commented out code?

Reply:
I will remove them